src/main/java/Parser.java

```diff
@@ -0,0 +1,156 @@
+public class Parser {
+    private static void checkValidDeadline (String command) throws WrongInputFormat, MissingEntries, MissingTaskName {
+        String[] commandWords = command.split("/");
+
+        if (commandWords.length != 2) {
+            throw new WrongInputFormat();
+        } else {
+            if (commandWords[0].length() < 10) {
+                throw new MissingTaskName();
+            }
+
+            if (commandWords[1].length() < 4) {
+                throw new MissingEntries();
+            }
+        }
+
+    }
+
+    private static void checkValidEvent (String command) throws WrongInputFormat, MissingEntries, MissingTaskName {
+        String[] commandWords = command.split("/");
+        if (commandWords.length != 3) {
+            throw new WrongInputFormat();
+        } else {
+
+            String eventName = commandWords[0].substring(6);
+            if (eventName.trim().isEmpty()) {
+                throw new MissingTaskName();
+            }
+
+            if (commandWords[1].length() < 6) {
+                throw new MissingEntries();
+            } else {
+                String from = commandWords[1].substring(5);
+            }
+
+            if (commandWords[2].length() < 4) {
+                throw new MissingEntries();
+            } else {
+                String to = commandWords[2].substring(3);
+            }
+
+        }
+    }
+
+    private static void checkValidTodo (String command) throws RedundantEntries, MissingTaskName {
+        String[] commandWords = command.split(" ");
+        if (commandWords.length < 2) {
+            throw new MissingTaskName();
+        } else if (commandWords.length > 2) {
+            throw new RedundantEntries();
+        }
+    }
+
+    private static boolean isValidDeadline (String command) {
+        try {
+            checkValidDeadline(command);
+        } catch (WrongInputFormat wif) {
+            System.out.println(UI.LINE_SEPARATOR);
+            System.out.println("OOPS! Wrong command format");
+            System.out.println(UI.LINE_SEPARATOR);
+            return false;
+        } catch (MissingEntries me) {
+            System.out.println(UI.LINE_SEPARATOR);
+            System.out.println("Missing deadline");
+            System.out.println(UI.LINE_SEPARATOR);
+            return false;
+        } catch (MissingTaskName mtn) {
+            System.out.println(UI.LINE_SEPARATOR);
+            System.out.println("Missing task name");
+            System.out.println(UI.LINE_SEPARATOR);
```

Comment:
Can consider abstracting this further to a printing method that takes in a string argument to avoid repeating yourself.